services/gossip/adapter/tcp/direct_transport.go

```diff
@@ -124,7 +127,8 @@ func newDirectTransportServer() *lockableTransportServer {
 
 func newLockableClientConnections() *lockableClientConnections {
 	return &lockableClientConnections{
-		peers: make(map[string]*clientConnection),
+		peers:  make(map[string]*clientConnection),
```

Comment:
@noambergIL note that I moved the peer list inside the lockable client connection, since it is accessed from two goroutines at the same time (topology update / client connection)